python/lsst/ts/externalscripts/auxtel/latiss_cwfs_align.py

```diff
@@ -462,35 +477,67 @@ async def run_cwfs(self):
         results_dict = self.calculate_results()
         return results_dict
 
+    def get_donut_region(self, center):
+
+        center_y = center[0]
+        center_x = center[1]
+        return (
+            center_y - self.side,
+            center_y + self.side,
+            center_x - self.side,
+            center_x + self.side,
+        )
+
+    def get_intra_donut_center(self):
+        return [
+            int(self.intra_result.brightestObjCentroidCofM[1]),
+            int(self.intra_result.brightestObjCentroidCofM[0]),
+        ]
+
+    def get_extra_donut_center(self):
+        return [
+            int(self.extra_result.brightestObjCentroidCofM[1]),
+            int(self.extra_result.brightestObjCentroidCofM[0]),
+        ]
+
     def create_donut_stamps_for_cwfs(self):
         """Create square stamps with donuts based on centroids."""
         # reset I1 and I2
         self.I1 = []
         self.I2 = []
 
-        ceny, cenx = int(self.intra_result.brightestObjCentroidCofM[1]), int(
-            self.intra_result.brightestObjCentroidCofM[0]
-        )
         self.log.debug(
-            f"Creating stamp for intra_image donut on centroid [y,x] = [{ceny},{cenx}] with a side "
+            f"Creating stamp for intra_image donut on centroid "
+            f"[y,x] = [{self.get_intra_donut_center()}] with a side "
             f"length of {2 * self.side} pixels"
         )
+
+        intra_box = self.get_donut_region(self.get_intra_donut_center())
+        print(intra_box)
+        print(intra_box[0])
         intra_square = self.intra_exposure.image.array[
-            ceny - self.side : ceny + self.side, cenx - self.side : cenx + self.side
+            intra_box[0] : intra_box[1], intra_box[2] : intra_box[3]
         ]
 
-        ceny, cenx = int(self.extra_result.brightestObjCentroidCofM[1]), int(
-            self.extra_result.brightestObjCentroidCofM[0]
-        )
+        if self.extra_focal_position_out_of_range:
```

Comment:
This if statement here needs to go to `get_extra_donut_center`. Here I'd expect to see only

```
extra_box = self.get_donut_region(self.get_extra_donut_center())
extra_square = self.extra_exposure.image.array[
extra_box[0] : extra_box[1], extra_box[2] : extra_box[3]
             ]
```